src/py-opentimelineio/opentimelineio/url_utils.py

```diff
@@ -56,17 +57,37 @@ def filepath_from_url(urlstr):
     # Parse provided URL
     parsed_result = urlparse.urlparse(urlstr)
 
+    # De-encode the parsed path
+    decoded_parsed_path = urlparse.unquote(parsed_result.path)
+
     # Convert the parsed URL to a path
-    filepath = PurePath(request.url2pathname(parsed_result.path))
+    filepath = PurePath(request.url2pathname(decoded_parsed_path))
 
     # If the network location is a window drive, reassemble the path
     if PureWindowsPath(parsed_result.netloc).drive:
-        filepath = PurePath(parsed_result.netloc + parsed_result.path)
+        filepath = PurePath(parsed_result.netloc + decoded_parsed_path)
+
+    # If the specified index is a windows drive, then append it to the other parts
+    elif PureWindowsPath(filepath.parts[0]).drive:
+        filepath = PurePosixPath(filepath.drive, *filepath.parts[1:])
 
-    # Otherwise check if the specified index is a windows drive, then offset the path
+    # If the specified index is a windows drive, then offset the path
     elif PureWindowsPath(filepath.parts[1]).drive:
         # Remove leading "/" if/when `request.url2pathname` yields "/S:/path/file.ext"
         filepath = PurePosixPath(*filepath.parts[1:])
 
+    # Should catch UNC paths,
+    # as parsing "file:///some/path/to/file.ext" doesn't provide a netloc
+    elif parsed_result.netloc and parsed_result.netloc != 'localhost':
+        # Paths of type: "file://host/share/path/to/file.ext" provide "host" as netloc
+        filepath = PurePath('//', parsed_result.netloc + decoded_parsed_path)
+
+        # When running `as_posix` the resulting path will have only 1 leading `/`,
+        # so we insert another `/` at the front of the string path and return it.
+        conformed_filepath = filepath.as_posix()
+        if conformed_filepath.startswith('/'):
+            conformed_filepath = '/' + conformed_filepath
```

Comment:
I think this test will always be `True`.
On Linux the `filepath` already has two leading slashes from the join above. Since `startwith('/')` will be true for 1+ slashes you get three.
Does Windows remove one of the slashes when joining or at the `as_posix()` call?

Reply:
Yes, that's what appears to be happening on my Windows machine even though the `filepath` was defined to start with `//`.

Reply:
Thanks for the context, I decided to re-work this a bit and it seems to be passing the tests now
```python
        conformed_filepath = filepath.as_posix()
        if not conformed_filepath.startswith('//'):
            conformed_filepath = '/' + conformed_filepath
```